installers/server/release/server.sh

```diff
@@ -124,20 +162,17 @@ fi
 if [ "$USE_URANDOM" != "false" ] && [ -e "/dev/urandom" ]; then
     SERVER_STARTUP_ARGS+=("-Djava.security.egd=file:/dev/./urandom")
 fi
-CMD="$JAVA_HOME/bin/java ${SERVER_STARTUP_ARGS[@]} -jar $SERVER_DIR/go.jar"
 
-echo "Starting Go Server with command: $CMD" >> $STDOUT_LOG_FILE
-echo "Starting Go Server in directory: $GO_WORK_DIR" >> $STDOUT_LOG_FILE
-cd "$SERVER_WORK_DIR"
+RUN_CMD=("$(autoDetectJavaExecutable)" ${SERVER_STARTUP_ARGS[@]} "-jar" "$SERVER_DIR/go.jar")
 
-if [ "$JAVA_HOME" == "" ]; then
-    echo "Please set JAVA_HOME to proceed."
-    exit 1
-fi
+echo "[$(date)] Starting Go Server with command: ${RUN_CMD[@]}" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] Starting Go Server in directory: $SERVER_WORK_DIR" >> $STDOUT_LOG_FILE
+cd "$SERVER_WORK_DIR"
 
 if [ "$DAEMON" == "Y" ]; then
-    eval exec nohup "$CMD" >> $STDOUT_LOG_FILE 2>&1 &
-    echo $! >$PID_FILE
+    "${RUN_CMD[*]}"  >> "$STDOUT_LOG_FILE" 2>&1 &
+    disown $!
+    echo $! >"$PID_FILE"
 else
-    eval exec "$CMD"
+    exec "${RUN_CMD[@]}"
```

Comment:
For context: There was a conversation about this, [here](https://gitter.im/gocd/gocd?at=5728cfae12cceadb7b1a38c5).